packages/core/state/selection/reducer.ts

```diff
@@ -169,14 +171,33 @@ export default makeReducer<SelectionStateBranch>(
             ...state,
             fileSelection: action.payload,
         }),
-        [SET_ANNOTATION_HIERARCHY]: (state, action) => ({
-            ...state,
-            annotationHierarchy: action.payload,
-            availableAnnotationsForHierarchyLoading: true,
+        [SET_ANNOTATION_HIERARCHY]: (state, action) => {
+            // combine recent hierarchy annotations with payload
+            const recentHierarchyAnnotationsUnfiltered = [
+                ...castArray(action.payload),
+                ...state.recentHierarchyAnnotations,
+            ];
 
-            // Reset file selections when annotation hierarchy changes
-            fileSelection: new FileSelection(),
-        }),
+            // remove duplicates and limit to 7 most recent
+            const recentHierarchyAnnotations = [
+                ...new Map(
+                    recentHierarchyAnnotationsUnfiltered.map((annotation) => [
+                        annotation.name,
+                        annotation,
+                    ])
+                ).values(),
+            ].slice(0, 7);
```

Comment:
Am I understanding that this will only preserve 7 columns? So if the user picks 8 columns the first one they picked will not be included when they re-open the explorer? Why not keep the entire last set of columns?